elastic/eui
# [EuiBasicTable][EuiInMemoryTable] Enable more action props to accept an optional callback + fix missing tooltips on collapsed actions

## Summary

@alvarezmelissa87 is working on a table that needs conditional action descriptions. She was previously attempting to use custom `render`ed actions to achieve this, but it would be much cleaner all around for their use case for EUI to simply allow `actions.description` to be a callback which we pass the current `item` to.

I also added this functionality to the `href` and `data-test-subj` keys while here, because why not/it seems useful! Example usage:

```tsx
<EuiBasicTable
  columns={[
    // ... columns
   {
      name: 'Actions',
      actions: [{
        name: (item) => `Delete ${item.name}`,
        description: (item) => item.someCondition ? 'This does X' : 'This does Y',
        href: (item) => `/delete/${item.id}`,
        'data-test-subj': (item) => `delete-${item.id}`,
      }],
    },
  ]}
/>
```

### Actions tooltip bugfix

While here, I noticed that the action descriptions (which normally renders as a delayed tooltip) was not correctly showing in the collapsed popover menu when they should be, so I added that in https://github.com/elastic/eui/pull/7373/commits/d29b623839b7501fb2256bfd0a5b87ca3e126e3d:

<img width="361" alt="" src="https://github.com/elastic/eui/assets/549407/c55ef212-af1d-43bb-a1ec-7dc0b63a8167">

![actions](https://github.com/elastic/eui/assets/549407/851e9656-8009-4fcb-9df8-dfd5aff7444f)

Doing this required adding a new `toolTipsProps` prop to `EuiContextMenu` so we could pass `delay="long"` to match the other actions tooltips (https://github.com/elastic/eui/pull/7373/commits/ba28dc07eb1381c0cc716100697ddd096ff8e47c)

As always, please [review by commit](https://github.com/elastic/eui/pull/7373/commits) (because I found a few more bugs even more bugs after that 🙈)

## QA

- Go to https://eui.elastic.co/pr_7373/#/tabular-content/tables#adding-actions-to-table
- Hover or focus on a non-disabled link action (online rows are enabled, offline rows are disabled)
- [x] Confirm the action description tooltip shows the row name
- Toggle the "Multiple Actions" switch in the top left corner
- [x] Confirm that keyboard tabbing into the actions column correctly makes the action buttons visible, unlike production
- [x] When clicking the "All actions" button, confirm that actions in the popover menu have tooltips, unlike production  

### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~
    ~- [ ] Checked in **mobile**~
- Docs site QA
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist
    ~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
